src/common/snippets/src/lowered/linear_ir.cpp

```diff
@@ -156,7 +156,8 @@ void LinearIR::debug_print(bool tds_as_pointers) const {
 
 void LinearIR::init_emitters(const std::shared_ptr<TargetMachine>& target) {
     for (auto& expr : m_expressions) {
-        if (!expr->get_emitter())
+        if (!(expr->get_emitter() ||
+            ov::is_type<snippets::op::Reshape>(expr->get_node())))
```

Comment:
To avoid these custom checks and conditions in the common code you may just to set `jit_nop_emitter` for `Reshape` in CPU generator that do nothing [as](https://github.com/openvinotoolkit/openvino/blob/058f0206e2076a7b580eef06120850ce79625072/src/plugins/intel_cpu/src/emitters/snippets/x64/cpu_generator.cpp#L73C7-L73C7) `Parameter`, `Result` and `Buffers`.
May I ask you to try out this solution please? Thanks!

Reply:
jit_nop_emitter is nice, updated. thanks Alexandra!